### PR TITLE
[FEAT]: Implemented Auth

### PR DESCRIPTION
**Pull request by shivanigangadharan**
- Added auth : login/signup/logout
- Connected tasks to logged in user
- Data also persists on refresh due to localStorage

### CODE REVIEW

**Review by dev-enforced — APPROVED** [review]
The functionality is working as mentioned in the PR!!Really good work on implementing the PR!! Code is also easy to understand.
Could implement persistent login too in the later stages with tasks and user details staying on the screen even after reload!!

File: src/App.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import TimerPage from './pages/timerPage/timerPage';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 import Settings from './pages/Settings/settings-page';
 import KnowMore from './pages/KnowMore/knowmore';
+import Login from './pages/login/login';
+import Signup from './pages/login/signup';
```
Comment (dev-enforced) [inline src/App.js:9]: Could use Barrel exports too making it easy for importing all the pages in a single line itself:
`import {Login,Signup,KnowMore,Settings} from "./pages"`

File: src/App.js [hunk 2]
```diff
@@ -15,6 +17,9 @@ function App() {
           <Route path="timerpage" element={<TimerPage />} />
           <Route path="settings" element={<Settings />} />
           <Route path="knowmore" element={<KnowMore />} />
+          <Route path="login" element={<Login />} />
+          <Route path="signup" element={<Signup />} />
```
Comment (dev-enforced) [inline src/App.js:21]: Covering the first 3 routes in a Private Route component  (meant only for logged in users ) related to TimerPage,Settings and KnowMore would be better!!

File: src/context/authContext.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            });
+            if (res.data.some(user => (user.email === email && user.password === password))) {
+                setLoggedUserID(res.data.find(user => user.email === email && user.password === password).id);
+                localStorage.setItem("user", res.data.find(user => user.email === email && user.password === password).id)
+                setUser(res.data.find(user => user.email === email && user.password === password));
+                return true;
+            } else {
+                console.log("No such user found");
+                return false;
+            }
```
Comment (dev-enforced) [inline src/context/authContext.js:44]: Liked this logic too!!

**Review by dev-enforced — APPROVED** [review]

File: src/context/authContext.js (hunk 3 above)
Comment (TejasShekar) [inline src/context/authContext.js:44]: Good implementation of userHandling. Liked it !

**Review by MonikaJhingan — APPROVED** [review]
The pr looks good to merged.

**Review by TejasShekar — APPROVED** [review]
Liked the use of clamp fn for width props in CSS.
UI is really good.

File: src/App.js (hunk 2 above)
Comment (shivanigangadharan) [reply src/App.js:21]: You're right! Thank you for this suggestion, will surely work on it.

**Pull request merged**